File: src/templates_py/templates_py/subscriber_template.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SubscriberTemplate(Node):

    def __init__(self, node_name: str) -> None:
        super().__init__(node_name)

        self.sub = self.create_subscription(
            msg_type=String,
            topic='/example_topic',
            callback=self.get_message,
            qos_profile=10
        )

    def get_message(self, msg: String) -> None:
        self.get_logger().info(message='Receiving: ' + msg.data)

def main(args=None):
    rclpy.init(args=args)
    subscriber_template = SubscriberTemplate(node_name='subscriber_template')
    rclpy.spin(node=subscriber_template)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/templates_cpp/src/subscriber_template.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

class SubscriberTemplate : public rclcpp::Node
{
    public:

    SubscriberTemplate() : Node("subscriber_template")
    {
        sub = this->create_subscription<std_msgs::msg::String>("example_topic", 10,
            std::bind(&SubscriberTemplate::callback, this, std::placeholders::_1)
        );
    }

    private:

    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub;

    void callback(const std_msgs::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "Receiving: %s", msg->data.c_str());
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto subscriber_template = std::make_shared<SubscriberTemplate>();
    rclcpp::spin(subscriber_template);
    rclcpp::shutdown();

    return 0;
}